redisinsight/ui/src/packages/redistimeseries-app/src/components/Chart/utils.ts

```diff
@@ -0,0 +1,264 @@
+        imgWidth * watermarkOpts.imageToWatermarkSizeRatio,
+        imgHeight * watermarkOpts.imageToWatermarkSizeRatio,
+      )
+
+      addImgToSVG(
+        svg,
+        watermarkOpts.image,
+        imgWidth - watermarkOpts.paddingRight - watermarkWidth,
+        imgHeight - watermarkOpts.paddingBottom - watermarkHeight,
+        watermarkHeight,
+        watermarkWidth,
+        watermarkOpts.opacity,
+      )
+      resolve(getDataURIFromSVGDocument(svg))
+      document.body.removeChild(watermarkCanvas)
+    })
+  )
+}
+
+
+
+function addImgToSVG(svg: Document, src: string, x: number, y: number, height: number, width: number, opacity = 1) {
+  const svgElement = svg.documentElement
+
+  const image = svg.createElementNS(svgElement.getAttribute('xmlns'), 'image')
+  image.setAttribute('xlink:href', src)
+  image.setAttribute('x', x.toString())
+  image.setAttribute('y', y.toString())
+  image.setAttribute('height', height.toString())
+  image.setAttribute('width', width.toString())
+  image.setAttribute('opacity', opacity.toString())
+
+  svg.firstChild!.appendChild(image)
+}
+
+function addTextToSVG(svg: Document, text: string, x: number, y: number, opacity = 1, style = '') {
+  const svgElement = svg.documentElement
+
+  const imgHeight = parseInt(svgElement.getAttribute('height')!)
+  const imgWidth = parseInt(svgElement.getAttribute('width')!)
+
+  const textEl = svg.createElementNS(svgElement.getAttribute('xmlns'), 'text')
+  textEl.innerHTML = text
+  textEl.setAttribute('x', (imgWidth + x).toString())
+  textEl.setAttribute('y', (imgHeight - y).toString())
+  textEl.setAttribute('fill-opacity', opacity.toString())
+  textEl.setAttribute('style', style)
+
+  svg.firstChild!.appendChild(textEl)
+}
+
+function createOffscreenCanvasForImage(image: DataURI): Promise<HTMLCanvasElement> {
+  const canvas = createOffscreenCanvas()
+  const ctx = canvas.getContext('2d')!
+  const img = new Image()
+
+  // https://stackoverflow.com/questions/22710627/tainted-canvases-may-not-be-exported/45129760#45129760
+  img.setAttribute('crossOrigin', 'anonymous')
+
+  return new Promise(resolve => {
+    img.addEventListener('load', function() {
+      canvas.height = img.height
+      canvas.width = img.width
+      ctx.drawImage(img, 0, 0)
+      resolve(canvas)
+    })
+    img.setAttribute('src', image)
+  })
+}
+
+function createOffscreenCanvas() {
+  const canvas = document.createElement('canvas')
+  canvas.style.position = 'absolute'
+  canvas.style.left = '-5000px'
+  document.body.appendChild(canvas)
+  return canvas
+}
+
+function getDataURIFromSVGDocument(svg: Document) {
+  return getDataURIFromSVGStr(new XMLSerializer().serializeToString(svg))
+}
+
+function getSVGDocumentFromDataURI(svgDataURI: DataURI) {
+  const svgStr = decodeURIComponent(svgDataURI.substr(SVG_URI_SCHEME.length))
+  return new DOMParser().parseFromString(svgStr, SVG_MIME_TYPE)
+}
+
+function getMimeTypeOfDataURI(uri: DataURI) {
+    return uri.split(/[:;,]/g)[1]
```

Comment:
Could you move RegExp to constants?
and please add Optional chaining operator before `[1]`, this is not safety
uri.split(...).?[1]